packages/orchestration/src/utils/denomHash.js

```diff
@@ -0,0 +1,30 @@
+// @ts-check
+import { sha256 } from '@noble/hashes/sha256';
```

Comment:
Watch out for bundler support for `"exports"` since this pattern will only work when the tools account for package exports.

Reply:
is there an alternative?

Reply:
Not with this package. It’s pretty narrowly prescribed:

https://github.com/paulmillr/noble-hashes/blob/main/package.json#L141-L145

That is, this package cannot be used with bundlers that have not caught up with package exports. There are very few such tools, but `node -r esm` is among them. Hopefully, there is no risk that this package will get caught up in that, but if you see integration errors to the tune of “I can’t import this”, this is why.

Reply:
**felgercarb!**

loadgen ci failed. :-/

now what do I do? sigh.

Reply:
We could fork or PR upstream to package export package relative paths to files with `.js` extensions. I’m not against forking this into Endo. We do that for Base64. We could harden the library. We would have recourse to fall through to native on XS if that becomes available.

Reply:
> I’m not against forking this into Endo. We do that for Base64.

I like that idea enough to reify it as an issue:

 - https://github.com/endojs/endo/issues/2335

I'll pursue other options in parallel, so it's not necessarily critical path.

Reply:
loadgen fixed with [58ff687](https://github.com/Agoric/agoric-sdk/pull/9576/commits/58ff687cd891732889554a007a78b21022ab0777)

Downstream imports won't get the patch but they don't need it.